dsgrid/registry/dataset_registry_manager.py

```diff
@@ -70,15 +73,41 @@ def _run_checks(self, config: DatasetConfig):
     def _check_dataset_consistency(self, config: DatasetConfig):
         spark = SparkSession.getActiveSession()
         path = Path(config.model.path)
-        load_data = read_dataframe(check_load_data_filename(path))
+
+#### Dan's change ###
+        load_data_df = read_dataframe(check_load_data_filename(path))
         load_data_lookup = read_dataframe(check_load_data_lookup_filename(path), cache=True)
         load_data_lookup = config.add_trivial_dimensions(load_data_lookup)
         with Timer(timer_stats_collector, "check_lookup_data_consistency"):
             self._check_lookup_data_consistency(config, load_data_lookup)
         with Timer(timer_stats_collector, "check_dataset_time_consistency"):
-            self._check_dataset_time_consistency(config, load_data)
+            self._check_dataset_time_consistency(config, load_data_df)
         with Timer(timer_stats_collector, "check_dataset_internal_consistency"):
-            self._check_dataset_internal_consistency(config, load_data, load_data_lookup)
+            self._check_dataset_internal_consistency(config, load_data_df, load_data_lookup)
+#### my change ####
+        data_schema_type = config.model.data_schema_type
+        if data_schema_type == DataSchemaType.STANDARD:
+            load_data = read_dataframe(check_load_data_filename(path))
+            load_data_lookup = read_dataframe(check_load_data_lookup_filename(path), cache=True)
+            load_data_lookup = config.add_trivial_dimensions(load_data_lookup)
+            with Timer(timer_stats_collector, "check_lookup_data_consistency"):
+                self._check_lookup_data_consistency(config, load_data_lookup)
+            with Timer(timer_stats_collector, "check_dataset_time_consistency_standard"):
+                self._check_dataset_time_consistency_standard(config, load_data)
+            with Timer(timer_stats_collector, "check_dataset_internal_consistency"):
+                self._check_dataset_internal_consistency(config, load_data, load_data_lookup)
+        elif data_schema_type == DataSchemaType.ONE_TABLE:
+            load_data_one_table = read_dataframe(
+                check_load_data_one_table_filename(path), cache=True
+            )
+            load_data_one_table = config.add_trivial_dimensions(load_data_one_table)
+            with Timer(timer_stats_collector, "check_one_table_data_consistency"):
+                self._check_one_table_data_consistency(config, load_data_one_table)
+            with Timer(timer_stats_collector, "check_dataset_time_consistency"):
+                self._check_dataset_time_consistency(config, load_data_one_table)
+        # else:
+        #     raise ValueError(f'data_schema_type={data_schema_type} not supported.')
+#### ####
```

Comment:
This checking code should probably be relegated to the DatasetConfig which could more naturally handle different schemas and dataset types.